wormhole-connect/src/utils/vaa.ts

```diff
@@ -184,12 +170,25 @@ export async function fetchVaaGuardian(
   const messageId = getEmitterAndSequence(txData);
   const { emitterChain, emitterAddress, sequence } = messageId;
 
-  const { vaaBytes: vaa } = await getSignedVAA(
-    WORMHOLE_RPC_HOSTS[0],
-    emitterChain,
-    emitterAddress,
-    sequence,
-  );
+  // round-robin through the RPC hosts
+  let vaa: Uint8Array | undefined;
+  for (const host of WORMHOLE_RPC_HOSTS) {
```

Comment:
my understanding of round robin is it's stateful across requests: if you have hosts `[A, B, C]`, the first request will go to `A`, then the second will go to `B`. this code will always try `A` first and only use `B` if `A` failed.

Reply:
Maybe we could simply add a module scoped variable that keeps track of the last index where the VAA was fetched successfully and keep reusing it on subsequent calls

Reply:
I understand this call is a fallback for when the explorer api doesn't have the VAA yet. I'm not sure the guardians will receive much pressure from here so it doesn't matter if it always try for A first, specially since the poll interval is of around 30s. Still, implementing the round robin should be straightforward.

Reply:
ok. so this is more for failover rather than load balancing?

Reply:
Yeah, we try to fetch the VAA from the wormholescan API first and if that doesn't have it, then we hit the guardians. If none of them have it then we wait 30 seconds and repeat this process.